drivers/sdhc/rcar_mmc.c

```diff
@@ -0,0 +1,2252 @@
+{
+	int ret = 0;
+	uint32_t reg;
+	struct mmc_rcar_data *data = dev->data;
+	struct sdhc_io ios = {0};
+
+	rcar_mmc_reset(dev);
+
+	/*  Disable SD clock (SD_CLK) output */
+	ret = rcar_mmc_enable_clock(dev, false);
+	if (ret) {
+		return ret;
+	}
+
+	/* set transfer data length to 0 */
+	rcar_mmc_write_reg32(dev, RCAR_MMC_SIZE, 0);
+
+	/* disable the SD_BUF read/write DMA transfer */
+	reg = rcar_mmc_read_reg32(dev, RCAR_MMC_EXTMODE);
+	reg &= ~RCAR_MMC_EXTMODE_DMA_EN;
+	rcar_mmc_write_reg32(dev, RCAR_MMC_EXTMODE, reg);
+	/* mask DMA irqs and clear dma irq flags */
+	rcar_mmc_reset_and_mask_irqs(dev);
+	/* set system address increment mode selector & 64-bit bus width */
+	reg = rcar_mmc_read_reg32(dev, RCAR_MMC_DMA_MODE);
+	reg |= RCAR_MMC_DMA_MODE_ADDR_INC | RCAR_MMC_DMA_MODE_WIDTH;
+	rcar_mmc_write_reg32(dev, RCAR_MMC_DMA_MODE, reg);
+
+	/* store version of of introductory IP */
+	data->ver = rcar_mmc_read_reg32(dev, RCAR_MMC_VERSION);
+	data->ver &= RCAR_MMC_VERSION_IP;
+
+	/*
+	 * set bus width to 1
+	 * timeout counter: SDCLK * 2^27
+	 * card detect time counter: SDϕ * 2^24
+	 */
+	reg = rcar_mmc_read_reg32(dev, RCAR_MMC_OPTION);
+	reg |= RCAR_MMC_OPTION_WIDTH_MASK | 0xEE;
+	rcar_mmc_write_reg32(dev, RCAR_MMC_OPTION, reg);
+
+	/* block count enable */
+	rcar_mmc_write_reg32(dev, RCAR_MMC_STOP, RCAR_MMC_STOP_SEC);
+	/* number of transfer blocks */
+	rcar_mmc_write_reg32(dev, RCAR_MMC_SECCNT, 0);
+
+	/*
+	 * SD_BUF0 data swap disabled.
+	 * Read/write access to SD_BUF0 can be performed with the 64-bit access.
+	 *
+	 * Note: when using the DMA, the bus width should be fixed at 64 bits.
+	 */
+	rcar_mmc_write_reg32(dev, RCAR_MMC_HOST_MODE, 0);
+	data->width_access_sd_buf0 = 8;
+
+	/* disable sampling clock controller, it is used for uhs/sdr104, hs200 and hs400 */
+	ret = rcar_mmc_disable_scc(dev);
+	if (ret) {
+		return ret;
+	}
+
+	/*
+	 * configure divider inside MMC controller
+	 * set maximum possible divider
+	 */
+	ios.clock = data->props.f_min;
+	rcar_mmc_set_clk_rate(dev, &ios);
+
+	data->restore_cfg_after_reset = 1;
+
+	return 0;
+}
+
+#ifdef CONFIG_RCAR_MMC_DMA_IRQ_DRIVEN_SUPPORT
+static void rcar_mmc_irq_handler(const void *arg)
+{
+	const struct device *dev = arg;
+
+	uint32_t dma_info1 = rcar_mmc_read_reg32(dev, RCAR_MMC_DMA_INFO1);
+	uint32_t dma_info2 = rcar_mmc_read_reg32(dev, RCAR_MMC_DMA_INFO2);
+
+	if (dma_info1 || dma_info2) {
+		struct mmc_rcar_data *data = dev->data;
+
+		rcar_mmc_write_reg32(dev, RCAR_MMC_DMA_INFO1_MASK, 0xfffffeff);
+		rcar_mmc_write_reg32(dev, RCAR_MMC_DMA_INFO2_MASK, ~0);
+		k_sem_give(&data->irq_xref_fin);
+	} else {
+		LOG_WRN("%s: warning: non-dma event triggers irq", dev->name);
+	}
+}
+#endif /* CONFIG_RCAR_MMC_DMA_IRQ_DRIVEN_SUPPORT */
+
+/* initialize and configure the Renesas MMC driver */
+static int rcar_mmc_init(const struct device *dev)
+{
+	int ret = 0;
+	struct mmc_rcar_data *data = dev->data;
+	const struct mmc_rcar_cfg *cfg = dev->config;
+
+	if (unlikely(cfg == NULL)) {
```

Comment:
Same question here- I don't think that `dev->config` can be a NULL pointer given that it is set statically